launcher/agent/agent.go

```diff
@@ -96,9 +98,28 @@ func (a *agent) Close() error {
 // MeasureEvent takes in a cel.Content and appends it to the CEL eventlog
 // under the attestation agent.
 func (a *agent) MeasureEvent(event cel.Content) error {
+	tpmDigestsmap, err := cel.GenerateDigestMap(defaultCELHashAlgo, event)
+	if err != nil {
+		return err
+	}
+
 	a.tpmMu.Lock()
 	defer a.tpmMu.Unlock()
-	return a.cosCel.AppendEvent(a.tpm, cel.CosEventPCR, defaultCELHashAlgo, event)
+
+	// extend TPM PCR digests
+	for hs, dgst := range tpmDigestsmap {
+		tpm2Alg, err := tpm2.HashToAlgorithm(hs)
+		if err != nil {
+			return err
+		}
+
+		if err := tpm2.PCRExtend(a.tpm, tpmutil.Handle(cel.CosEventPCR), tpm2Alg, dgst, ""); err != nil {
+			return fmt.Errorf("failed to extend event to PCR%d: %v", cel.CosEventPCR, err)
+		}
+	}
+
+	// append TPM PCR event to eventlog
+	return a.cosCel.AppendEvent(cel.CosEventPCR, event, tpmDigestsmap)
```

Comment:
MeasureEvent only appends TPM PCR event, do u have a plan to have support for measuring RTMR event to event log and integration into the current container runner for TDX CS image?

Reply:
I think it is done in a different PR. https://github.com/google/go-tpm-tools/pull/478/

Reply:
AppendEvent is now register-agnostics, then actual "extend" operation on pcr or rtmr should be done outside "AppendEvent" function.

Reply:
nit: Given that `MeasureEvent` appends TPM PCR event, could you rename it to something like `MeasurePCREvent` to distinguish it from RMTR event measurement?